unstructured_ingest/v2/processes/connectors/gitlab.py

```diff
@@ -0,0 +1,249 @@
+        # If a scheme or netloc are provided, use the parsed base url
+        if parsed_gh_url.scheme or parsed_gh_url.netloc:
+            values["base_url"] = f"{parsed_gh_url.scheme}://{parsed_gh_url.netloc}"
+        values["repo_path"] = parsed_gh_url.path
+        while values["repo_path"].startswith("/"):
+            values["repo_path"] = values["repo_path"][1:]
+
+        return values
+
+    @SourceConnectionError.wrap
+    @requires_dependencies(["gitlab"], extras="gitlab")
+    def get_project(self) -> "Project":
+        from gitlab import Gitlab
+
+        logger.info(f"Accessing Base URL: '{self.base_url}'", extra={"base_url": self.base_url})
+
+        gitlab = Gitlab(self.base_url, private_token=self.access_config.access_token)
+
+        logger.info(f"Accessing Project: '{self.repo_path}'", extra={"repo_path": self.repo_path})
+
+        project = gitlab.projects.get(self.repo_path)
+
+        logger.info(
+            f"Successfully accessed project '{self.repo_path}'", extra={"repo_path": self.repo_path}
+        )
+        return project
+
+
+class GitLabIndexerConfig(IndexerConfig):
+    pass
+
+
+@dataclass
+class GitLabIndexer(Indexer):
+    connection_config: GitLabConnectionConfig
+    index_config: GitLabIndexerConfig
+    registry_name: str = "gitlab"
+
+    @staticmethod
+    def is_file_type_supported(path: str) -> bool:
+        # Workaround to ensure that auto.partition isn't fed with .yaml, .py, etc. files
+        # TODO: What to do with no filenames? e.g. LICENSE, Makefile, etc.
+        supported = path.endswith(
+            (
+                ".md",
+                ".txt",
+                ".pdf",
+                ".doc",
+                ".docx",
+                ".eml",
+                ".heic",
+                ".html",
+                ".png",
+                ".jpg",
+                ".ppt",
+                ".pptx",
+                ".xml",
+            ),
+        )
+        if not supported:
+            logger.debug(
+                f"The file {path!r} is discarded as it does not contain a supported filetype.",
+            )
+        return supported
+
+    @requires_dependencies(["gitlab"], extras="gitlab")
+    def precheck(self) -> None:
+        from gitlab import Gitlab
+        from gitlab.exceptions import GitlabError
+
+        try:
+            gitlab = Gitlab(
+                self.connection_config.base_url,
+                private_token=self.connection_config.access_config.access_token,
+            )
+            if self.connection_config.access_config.access_token is not None:
+                gitlab.auth()
+            else:
+                gitlab.projects.get(self.connection_config.repo_path)
+        except GitlabError as gitlab_error:
+            logger.error(f"failed to validate connection: {gitlab_error}", exc_info=True)
+            raise SourceConnectionError(f"failed to validate connection: {gitlab_error}")
+
+    def does_path_match_glob(self, path: str) -> bool:
```

Comment:
Same comment as [here](https://github.com/Unstructured-IO/unstructured-ingest/pull/157#discussion_r1801698776). We don't need the connector specific glob filter anymore.